### PR TITLE
Bjorncs/tls certificate validation

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

**Comment by bjorncs** [comment]
I have currently modelled the authorization mode (`PeerAuthorizerTrustManager.Mode`) as a separate parameter. We might want to add it to the config file as a separate field instead.

**Review by vekterli — APPROVED** [review]
:+1: a couple of minor questions

File: security-utils/src/main/java/com/yahoo/security/SslContextBuilder.java [hunk 1]
```diff
@@ -106,11 +111,16 @@ public SSLContext build() {
         }
     }
 
-    private static TrustManager[] createTrustManagers(KeyStoreSupplier trustStoreSupplier)
+    private static TrustManager[] createTrustManagers(TrustManagersFactory trustManagersFactory, KeyStoreSupplier trustStoreSupplier)
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply security-utils/src/main/java/com/yahoo/security/SslContextBuilder.java]: I did not bother making two interfaces with identical signature ;). They have private accessibility anyway.

File: security-utils/src/main/java/com/yahoo/security/tls/authz/PeerAuthorizerTrustManager.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                wrappedManagers[i] = managers[i];
+            }
+        }
+        return wrappedManagers;
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        defaultTrustManager.checkClientTrusted(chain, authType);
+        authorizePeer(chain[0], authType, true, null);
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply security-utils/src/main/java/com/yahoo/security/tls/authz/PeerAuthorizerTrustManager.java]: Yes. The `X509ExtendedTrustManager` interface requires all implementation to throw an `IllegalArgumentException` under that scenario: "if null or zero-length array is passed in for the chain parameter or if null or zero-length string is passed in for the authType parameter". Calling `defaultTrustManager. checkClient/ServerTrusted ` will therefore ensure that accessing `chain[0]` later on is safe.

**Comment by bjorncs** [comment]
FYI @havardpe

**Pull request merged**